### ISSUE TITLE
_extern_path in fileclient is broken

### ISSUE THREAD

**Issue by nasenbaer13** [saltstack/salt#36371]
### Description of Issue/Question

_extern_path constructs a path for the file cache but neglects url parts like query strings, fragments and so on.

### Setup

Consider a state like:
```
/target/dir/geoipDB.tar.gz:
  file.managed:
    - makedirs: True
    - source: "https://download.maxmind.com/app/geoip_download?edition_id=some-db&suffix=tar.gz&license_key=secret_key"
    - source_hash: "https://download.maxmind.com/app/geoip_download?edition_id=some-db&suffix=tar.gz.md5&license_key=secret_key"
```

This state fails because the url in `source` and `source_hash` are parsed and map to the same file. For determining the hash, the actual file is copied over the file containing the hash. Determining the hash then fails.

### Steps to Reproduce Issue

Create a state with sources only distinct by their query string.

### Versions Report
Custom salt based on 2016.3 but the actual bug is still present in https://github.com/saltstack/salt/blob/develop/salt/fileclient.py#L712

### Possible solution:

append query strings, fragments etc instead of `url_data.path` only. I can prepare a patch if this sounds reasonable.

**Comment by gtmanfred** [comment]
Can you check if this pr fixes your issue?

https://github.com/saltstack/salt/pull/36305

Thanks

**Comment by rallytime** [comment]
Since this is fixed via #36305, I'll close this.
